creator-node/src/routes/files.js

```diff
@@ -526,11 +578,18 @@ const getDirCID = async (req, res) => {
         )
       )
     }
-    storagePath = queryResults.storagePath
+
+    storagePath = DiskManager.computeFilePathInDir(
+      dirCID,
+      queryResults.multihash
+    )
     redisClient.set(cacheKey, storagePath, 'EX', FILE_CACHE_EXPIRY_SECONDS)
   }
 
-  // Attempt to stream file to client
+  // CID is the file CID, parse it from the storagePath
+  CID = storagePath.split('/').slice(-1).join('')
```

Comment:
this is probably unrelated since this was already there but why do we compute CID from storagePath instead of db record file.multihash?

Reply:
right, makes sense